### PR TITLE
Documentation

### LINKED ISSUE #111: Fix README
Things to look for:

- [x] examples still referencing old config object in route setup (routeId)
- [x] specificy `context.config` is on both reply and request
- [x] specify `getLabel`  should not have a `.` at the beginning and the end of the string.
- [ ] specify that `getLabel` is needed in dynamic mode
- [ ] add a migration guide

### PR DESCRIPTION
**Pull request by dnlup**
## 🚨 Proposed changes

> Please review the [guidelines for contributing](../../CONTRIBUTING.md) to this repository.

[[Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.]]

In this PR I am finishing aligning the documentation to the latest code changes and adding a migration guide and an examples folder.

Closes #111

## ⚙️ Types of changes

What types of changes does your code introduce? _Put an `x` in the boxes that apply_

-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [x] Documentation Update (if none of the other choices apply)
-   [ ] Refactor

### CODE REVIEW

**Comment by dnlup** [comment]
TODO:

- [ ] migration guide
- [ ] examples folder with common use cases

**Comment by simonecorsi** [comment]
lgtm so far! 💪

File: README.md [hunk 1]
```diff
@@ -331,7 +346,53 @@ The `getLabel` function in this mode will have the following signature:
 -   `reply`
 -   **Returns:** <`string`> The route label string without any `.` at the beginning or end.
 
-The `this` context of the function is bound to the fastify instance of the request.
+The `this` context of the function is bound to the fastify instance of the request. Pay attention to avoid returing empty strings or strings with leading and trailing `.`. Also, don't use arrow functions otherwhise the `this` context won't refer to the fastify instance.
+
+The default function used ifn you don't pass your custom one returns the same string that is computed in `static` mode, so the `dynamic` mode is not very useful if you don't define yuor own `getLabel` function.
```
Comment (antoniomuso) [inline README.md:351]: `ifn` -> `if`; 
 I don't know, but it seems to miss some comma in this sentence.

Comment (antoniomuso) [reply README.md:351]: WDYT to rephrase in something like this:
If you don't pass your custom one, the default function returns the exact string computed in `static` mode ...

Comment (dnlup) [reply README.md:351]: Yes that sounds better, thanks

File: README.md [hunk 2]
```diff
@@ -288,14 +301,14 @@ There are two different mode to generate the label for each route that the plugi
...
--   the routes prefix passed in the plugin option `collect.routes.prefix` (`routesPrefix`)
+-   the [`fastify` prefix](https://www.fastify.io/docs/latest/Reference/Server/#prefix) used to register the plugin (normalized replacing `/` with `.`), we call it `fastifyPrefix`
+-   the routes prefix passed to the plugin option `routes.prefix`, we call it `routesPrefix`
 
 Generating a computed prefix like this:
 
 `<fastifyPrefix>.<routesPrefix>`
 
-##### `static`
+##### `static` mode
```
Comment (antoniomuso) [inline README.md:311]: I think it is so small, WYT if we increase the size?

Comment (dnlup) [reply README.md:311]: Yeah I think so too, but I wanted to have these section nested in the TOC (which is generated automatically by header size). I am not sure how to fix it atm.

Comment (antoniomuso) [reply README.md:311]: Ce can put `Routes labels generation mode` at the same level of `Configuration options` WDYT? Or we can remove substitute `API` directly with `Configuration options`

Comment (dnlup) [reply README.md:311]: 🤔 I prefer the first solution

Comment (antoniomuso) [reply README.md:311]: ok :+1:

**Comment by antoniomuso** [comment]
LGTM, Good works!

**Review by antoniomuso — APPROVED** [review]

**Review by simonecorsi — APPROVED** [review]
I really liked the addition of examples! GJ! 🚀

**Pull request merged**